### PR TITLE
i#2534 Ignore i#1702: Fix {memtrace,instrace}_simple

### LINKED ISSUE #2534: memtrace_simple and instrace_simple don't need to ignore predicated instructions
Both `memtrace_simple` and `instrace_simple` will not place clean calls in front of predicated instructions, because of #1702, however these samples don't actually have to deal with predicated clean calls so they are not applicable.

We can remove the issue reference and the associated instruction predication test.

### PR DESCRIPTION
**Pull request by toshipiazza**
memtrace and instrace both mention i#1702 as a reason not to insert a
clean call in front of a predicated instruction. However, this issue is
only applicable to clean calls which wish to reuse the predication of
the instruction.

Fixes #2534

### CODE REVIEW

File: api/samples/instrace_simple.c [hunk 1]
```diff
@@ -234,7 +229,7 @@ event_app_instruction(void *drcontext, void *tag, instrlist_t *bb,
          * Using a fault to handle a full buffer should be more robust, and the
          * forthcoming buffer filling API (i#513) will provide that.
          */
-        IF_AARCHXX(&& !instr_is_exclusive_store(instr)))
+        IF_AARCHXX_ELSE(!instr_is_exclusive_store(instr), true))
```
Comment (derekbruening) [inline api/samples/instrace_simple.c]: suggest: IMHO it's nicer to put the && inside rather than having the else true, here and below

Comment (toshipiazza) [inline api/samples/instrace_simple.c]: Done here, though below there's now only one condition in the if statement, so this doesn't easily apply, right?

Comment (derekbruening) [inline api/samples/instrace_simple.c]: Oh, right.  I guess I'd put the whole if inside an ifdef (just not a fan of if (true)) but your call.

Comment (derekbruening) [inline api/samples/instrace_simple.c]: Scratch that if it would put the {} in ifdefs which is even worse IMHO

Comment (toshipiazza) [inline api/samples/instrace_simple.c]: Ack

**Pull request merged**